src/Analyzers/CSharp/Tests/MakeDeclarationPartial/MakeDeclarationPartialTests.cs

```diff
@@ -0,0 +1,234 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.MakeDeclarationPartial;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.CodeAnalysis.CSharp.CSharp.UnitTests.MakeDeclarationPartial
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsMakeDeclarationPartial)]
+    public sealed class MakeDeclarationPartialTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        public MakeDeclarationPartialTests(ITestOutputHelper logger)
+            : base(logger)
+        {
+        }
+
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new CSharpMakeDeclarationPartialCodeFixProvider());
```

Comment:
can you update to new test framework.  i'll review tests then.  thanks.

Reply:
look for usages of VerifyCS in this project :)

Reply:
@DoctorKrolic See https://github.com/dotnet/roslyn/blob/main/src/Analyzers/CSharp/Tests/FixIncorrectConstraint/FixIncorrectConstraintTests.cs

Reply:
VB side doesn't have a `TestState` and `FixedState`. Is it possible to test VB with several documents?

Reply:
@sharwell ?

Reply:
@DoctorKrolic It does have `TestState` and `FixedState`. See for example:

https://github.com/dotnet/roslyn/blob/208c7da2a7e3175e1a7cab42ff6742ed19d5d130/src/EditorFeatures/VisualBasicTest/InlineMethod/VisualBasicInlineMethodTests.vb#L29-L32

Reply:
I was trying to access them using `With { ... }` syntrax and they didn't show up. Weird...

Reply:
@Youssef1313 VB has a bit different approach for test with several documents, I see... Thanks!